src/Symfony/Component/HttpKernel/Fragment/AbstractSurrogateFragmentRenderer.php

```diff
@@ -92,4 +96,17 @@ private function generateSignedFragmentUri($uri, Request $request)
 
         return substr($fragmentUri, strlen($request->getSchemeAndHttpHost()));
     }
+
+    private function containsNonScalars(array $values)
+    {
+        foreach ($values as $key => $value) {
+            if (is_array($value)) {
+                return $this->containsNonScalars($value);
+            } elseif (!is_scalar($value) && null !== $value) {
```

Comment:
[`is_scalar()` returns `false` with `null`](https://3v4l.org/GEBkU). We want to support nulls hence the second part of the condition.